template-parts/calendar/single-event.php

```diff
@@ -70,88 +41,120 @@
-			<?php if ( $event['contactOrganization'] && ! empty( $labels['contact_organization'] ) ) : ?>
+			<?php if ( ! empty( $event['contactOrganization'] ) && ! empty( $labels['contact_organization'] ) ) : ?>
 				<dt class="single-event-contact-org-label"><?php echo esc_html( $labels['contact_organization'] ); ?></dt>
 				<dd class="single-event-contact-org-info"><?php echo wp_kses_post( $event['contactOrganization'] ); ?></dd>
 			<?php endif; ?>
-			<?php if ( $event['contact_name'] && ! empty( $labels['contact_name'] ) ) : ?>
+			<?php if ( ! empty( $event['contact_name'] ) && ! empty( $labels['contact_name'] ) ) : ?>
 				<dt class="single-event-contact-name-label"><?php echo esc_html( $labels['contact_name'] ); ?></dt>
 				<dd class="single-event-contact-name-info"><?php echo wp_kses_post( $event['contact_name'] ); ?></dd>
 			<?php endif; ?>
-			<?php if ( $event['contact_email'] && ! empty( $labels['contact_email'] ) ) : ?>
+			<?php if ( ! empty( $event['contact_email'] ) && ! empty( $labels['contact_email'] ) ) : ?>
 				<dt class="single-event-contact-email-label"><?php echo esc_html( $labels['contact_email'] ); ?></dt>
 				<dd class="single-event-contact-email-info"><?php printf( '<a href="mailto:%s" class="single-event-contact-email-link">%s</a>', esc_url( $event['contact_email'] ), esc_html( $event['contact_email'] ) ); ?></dd>
 			<?php endif; ?>
-			<?php if ( $event['phone'] && ! empty( $labels['contact_phone'] ) ) : ?>
+			<?php if ( ! empty( $event['phone'] ) && ! empty( $labels['contact_phone'] ) ) : ?>
 				<dt class="single-event-contact-phone-label"><?php echo esc_html( $labels['contact_phone'] ); ?></dt>
 				<dd class="single-event-contact-phone-info"><?php echo wp_kses_post( $event['phone'] ); ?></dd>
 			<?php endif; ?>
+			<?php
+			// Check if this calendar has custom fields.
+			$calendar_id = responsive_calendar_get_calendar_id();
+			if ( $buCalendar->hasCustomFields( $calendar_id ) ) {
+
+				/**
+				 * Retrieve the fields, will be cached from hasCustomFields call.
+				 *
+				 * Note that these are not fields associated with data/values,
+				 * rather these are the base fields this calendar has. The
+				 * actual event, e.g. $event, contains the data/value for an
+				 * event's custom fields.
+				 */
+				$fields = $buCalendar->getCustomFields( $calendar_id );
+
+				// Now we loop through each field and check
+				// if that field has a value for this event.
+				foreach ( $fields as $field ) {
+
+					// Stores the value of the custom field.
+					$event_field = $event[ $field['name'] ];
 
+					// If the event has this field and it has a value then print
+					// the field label and the event's field value.
+					if ( ! empty( $event[ $field['name'] ] ) && ! empty( $event_field ) ) {
+						?>
+						<dt class="single-event-custom-field-label"><?php echo esc_html( $field['label'] ); ?></dt>
+						<dd class="single-event-custom-field-info"><?php echo wp_kses_post( $event_field ); ?></dd>
```

Comment:
@Adam42 : How do you envision leveraging the `html_type` data for each event? Is that to tell us how to sanitize the data for output?

Reply:
@toddmilliken The primary use of `html_type` is in the Add/Edit event view in the admin section of bu.edu/phpbin/calendar/app. It's used there to figure out how to render a custom field to the admin user.

So on the API consuming side, here WP, it's not of much use except to, as you noted, guide us on what kind of data the custom field may contain so we can decide how to sanitize and render it.

Since right now all we have is `textarea` and `checkbox` what that amounts to is plain text/HTML [ long story short, BU Calendar admin allows `<b>`, `<i>` or `<p>` tags in select fields but otherwise no HTML although we do have an enhancement that one day will allow any HTML ] or a boolean type value.

So far, I think checkbox use has been limited to values used by the theme to display or not display something. Meaning I think we could also potentially treat `checkbox` as something to not display at all. I believe Alana's use of checkbox was so a calendar admin could tag an event which Alana would then use to determine if the theme should display in a different way. In other words, I don't think the value of `checkbox` is something that we will typically want to display to end users [ although I could be wrong and we should at least sanitize it as if we would display it ].

Reply:
Ah ok, I'll prevent it from outputting if checkbox is set!

Reply:
^ Added to line 144 to skip checkbox fields.